---
title: 'My first ipynb'
subtitle: 'this is kind of fun!'
author: Ran Li
---

# Intro

My first .pynb quarto doc.

Let's show how quarto code chunks are 100% translatable from quarto markdown (.qmd) to Jupyter notebooks (.jupyter). Below we will do code chunk that has a sumamry and is code folded.

In [4]:
#| code-fold: true
#| code-summary: "This is some highly complex code for computation"

print(1+29392)

29393


# Data

Okay let's do the penguins thing but in Python! 

## Raw data

Starting with the data.

In [13]:
#| code-fold: false

import polars as pl

df = pl.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/refs/heads/master/penguins.csv')

print(df.shape)
print(df.head())

(344, 7)
shape: (5, 7)
┌─────────┬───────────┬────────────────┬───────────────┬───────────────────┬─────────────┬────────┐
│ species ┆ island    ┆ bill_length_mm ┆ bill_depth_mm ┆ flipper_length_mm ┆ body_mass_g ┆ sex    │
│ ---     ┆ ---       ┆ ---            ┆ ---           ┆ ---               ┆ ---         ┆ ---    │
│ str     ┆ str       ┆ f64            ┆ f64           ┆ i64               ┆ i64         ┆ str    │
╞═════════╪═══════════╪════════════════╪═══════════════╪═══════════════════╪═════════════╪════════╡
│ Adelie  ┆ Torgersen ┆ 39.1           ┆ 18.7          ┆ 181               ┆ 3750        ┆ MALE   │
│ Adelie  ┆ Torgersen ┆ 39.5           ┆ 17.4          ┆ 186               ┆ 3800        ┆ FEMALE │
│ Adelie  ┆ Torgersen ┆ 40.3           ┆ 18.0          ┆ 195               ┆ 3250        ┆ FEMALE │
│ Adelie  ┆ Torgersen ┆ null           ┆ null          ┆ null              ┆ null        ┆ null   │
│ Adelie  ┆ Torgersen ┆ 36.7           ┆ 19.3          ┆ 193               ┆ 


So this original dataframe was 344 rows - some of which has missling values.

## Clean

Let's do some basic cleaning which is just to remove the rows with all missing values.



In [10]:
df_clean = df.drop_nulls()
df_clean.shape

(333, 7)

Now we can see that the cleaned dataframe which has any row with NULL data in any column was removed only has 333 rows - we dropped 11 rows. Looks good. I think we can start our analysis. 

# Analysis

## Plot